src/controllers/csv-processor.ts

```diff
@@ -124,10 +106,30 @@ function setupPagination(page: number, total_pages: number): Array<string | numb
     return pages;
 }
 
-export const processCSV = async (filename: string, page: number, size: number): Promise<ProcessedCSV> => {
-    const dataLateService = new DataLakeService();
+export const processCSV = async (dataset: Dataset, page: number, size: number): Promise<ViewErrDTO | ViewDTO> => {
+    const datalakeService = new DataLakeService();
     try {
-        const buff = await dataLateService.downloadFile(`${filename}.csv`);
+        const datafiles = await dataset.datafiles;
+        const datafile: Datafile | undefined = datafiles
+            .sort(
+                (first: Datafile, second: Datafile) =>
+                    new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
```

Comment:
It comes from the database... When the first entry is created it automatically sets the date and time for the creation date.  I did this so as you say in case of disaster recovery.